### PR TITLE
UOE-11183: consume StoreURL from New portal UI and forward appstoreurl and appsource for applovinmax

### PR DESCRIPTION
**Pull request by AvinashKapre**
**func extractItunesIdFromAppStoreUrl benchmarking**

**Wthout Regex:**
cpu: 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
BenchmarkFindTrailingDigits-8            5594328               201.0 ns/op            48 B/op          9 allocs/op
BenchmarkFindTrailingDigits-8            5764165               204.2 ns/op            48 B/op          9 allocs/op
BenchmarkFindTrailingDigits-8            5838740               205.0 ns/op            48 B/op          9 allocs/op
BenchmarkFindTrailingDigits-8            5772484               198.7 ns/op            48 B/op          9 allocs/op
BenchmarkFindTrailingDigits-8            5722549               205.3 ns/op            48 B/op          9 allocs/op
BenchmarkFindTrailingDigits-8            5892768               201.9 ns/op            48 B/op          9 allocs/op
BenchmarkFindTrailingDigits-8            5656333               208.1 ns/op            48 B/op          9 allocs/op
BenchmarkFindTrailingDigits-8            5931055               202.3 ns/op            48 B/op          9 allocs/op
BenchmarkFindTrailingDigits-8            5712402               206.9 ns/op            48 B/op          9 allocs/op
BenchmarkFindTrailingDigits-8            5530390               211.6 ns/op            48 B/op          9 allocs/op

**With Regex**
cpu: 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
BenchmarkExtractItunesIdFromAppStoreUrl-8        1000000              1017 ns/op              32 B/op          1 allocs/op
BenchmarkExtractItunesIdFromAppStoreUrl-8        1210808               989.0 ns/op            32 B/op          1 allocs/op
BenchmarkExtractItunesIdFromAppStoreUrl-8        1209648               998.0 ns/op            32 B/op          1 allocs/op
BenchmarkExtractItunesIdFromAppStoreUrl-8        1206888               982.3 ns/op            32 B/op          1 allocs/op
BenchmarkExtractItunesIdFromAppStoreUrl-8        1182054              1004 ns/op              32 B/op          1 allocs/op
BenchmarkExtractItunesIdFromAppStoreUrl-8        1209626              1044 ns/op              32 B/op          1 allocs/op
BenchmarkExtractItunesIdFromAppStoreUrl-8        1179132              1015 ns/op              32 B/op          1 allocs/op
BenchmarkExtractItunesIdFromAppStoreUrl-8        1000000              1028 ns/op              32 B/op          1 allocs/op
BenchmarkExtractItunesIdFromAppStoreUrl-8        1000000              1010 ns/op              32 B/op          1 allocs/op
BenchmarkExtractItunesIdFromAppStoreUrl-8        1206337              1050 ns/op              32 B/op          1 allocs/op

### CODE REVIEW

File: modules/pubmatic/openwrap/beforevalidationhook.go [hunk 1]
```diff
@@ -530,6 +531,10 @@ func (m OpenWrap) handleBeforeValidationHook(
 		if impExt.Wrapper != nil {
 			adserverURL = impExt.Wrapper.AdServerURL
 		}
+
+		if rCtx.Endpoint == models.EndpointAppLovinMax && payload.BidRequest.App != nil && payload.BidRequest.App.StoreURL == "" {
+			rCtx.AppLovinMax = updateProfileAppStoreUrl(rCtx, payload.BidRequest, impExt)
```
Comment (pm-isha-bharti) [inline modules/pubmatic/openwrap/beforevalidationhook.go:536]: Can we return just the storeURL here?

File: modules/pubmatic/openwrap/beforevalidationhook.go [hunk 2]
```diff
@@ -1289,3 +1300,50 @@ func isValidURL(urlVal string) bool {
 	}
 	return validator.IsRequestURL(urlVal) && validator.IsURL(urlVal)
 }
+
+func updateProfileAppStoreUrl(rctx models.RequestCtx, bidRequest *openrtb2.BidRequest, impExt *models.ImpExtension) models.AppLovinMax {
+	appStoreUrl := rctx.PartnerConfigMap[models.VersionLevelConfigID][models.AppStoreUrl]
+	if appStoreUrl == "" {
+		glog.Errorf("[AppLovinMax] [PubID]: %d [ProfileID]: %d [Error]: app storeurl not prsent in DB", rctx.PubID, rctx.ProfileID)
```
Comment (pm-isha-bharti) [inline modules/pubmatic/openwrap/beforevalidationhook.go:1307]: nitpick: type present

File: modules/pubmatic/openwrap/beforevalidationhook.go [hunk 3]
```diff
@@ -533,7 +533,7 @@ func (m OpenWrap) handleBeforeValidationHook(
 		}
 
 		if rCtx.Endpoint == models.EndpointAppLovinMax && payload.BidRequest.App != nil && payload.BidRequest.App.StoreURL == "" {
-			rCtx.AppLovinMax = updateProfileAppStoreUrl(rCtx, payload.BidRequest, impExt)
+			rCtx.AppLovinMax.AppStoreUrl = getProfileAppStoreUrlAndUpdateItunesID(rCtx, payload.BidRequest, impExt)
```
Comment (pm-isha-bharti) [inline modules/pubmatic/openwrap/beforevalidationhook.go:536]: Ideally, getProfileAppStoreUrl and updateSkadnSourceapp method should be separate, we can do the refactoring later if it takes time.

Comment (pm-isha-bharti) [reply modules/pubmatic/openwrap/beforevalidationhook.go:536]: Please link the ticket to the UOE-11183

**Review by pm-isha-bharti — APPROVED** [review]

File: modules/pubmatic/openwrap/beforevalidationhook.go (hunk 3 above)
Comment (AvinashKapre) [reply modules/pubmatic/openwrap/beforevalidationhook.go:536]: https://inside.pubmatic.com:9443/jira/browse/UOE-11191

**Pull request merged**